### PR TITLE
SMP

### PR DESCRIPTION
**Pull request by N00byEdge**

### CODE REVIEW

File: src/platform/x86_64/x86_64.zig [hunk 1]
```diff
@@ -41,8 +41,36 @@ pub fn platform_init() !void {
...
+  os.memory.paging.CurrentContext.apply();
+  try interrupts.init_interrupts();
+  setup_gdt();
+
+  asm volatile("sti");
+}
+
+var bsp_task: os.thread.Task = .{};
+
+pub fn self_exited() !?*os.thread.Task {
```
Comment (notYuriy) [inline src/platform/x86_64/x86_64.zig:62]: Why error here?

File: src/platform/x86_64/x86_64.zig [hunk 2]
```diff
@@ -372,8 +406,8 @@ pub fn inl(port: u16) u32 {
 
 pub fn debugputch(ch: u8) void {
   outb(0xe9, ch);
-  serial.port(1).write(ch);
-  serial.port(2).write(ch);
-  serial.port(3).write(ch);
-  serial.port(4).write(ch);
+  // serial.port(1).write(ch);
```
Comment (notYuriy) [inline src/platform/x86_64/x86_64.zig:409]: This can be safely uncommented

**Review by notYuriy — CHANGES_REQUESTED** [review]

File: src/platform/x86_64/x86_64.zig (hunk 1 above)
Comment (N00byEdge) [reply src/platform/x86_64/x86_64.zig:62]: For flexibility for the implementation, but I guess that should be a `@panic()` instead

**Review by notYuriy — APPROVED** [review]

**Comment by notYuriy** [comment]
LGTM

**Pull request merged**